internal/dispatch/graph/reachableresources_test.go

```diff
@@ -603,3 +609,109 @@ func TestCaveatedReachableResources(t *testing.T) {
 		})
 	}
 }
+
+func TestReachableResourcesWithConsistencyLimitOf1(t *testing.T) {
+	defer goleak.VerifyNone(t, goleakIgnores...)
+
+	ctx, dispatcher, revision := newLocalDispatcherWithConsistencyLimit(t, 1)
+	defer dispatcher.Close()
+
+	target := ONR("user", "owner", "...")
+	stream := dispatch.NewCollectingDispatchStream[*v1.DispatchReachableResourcesResponse](ctx)
+	err := dispatcher.DispatchReachableResources(&v1.DispatchReachableResourcesRequest{
+		ResourceRelation: RR("folder", "view"),
+		SubjectRelation: &core.RelationReference{
+			Namespace: target.Namespace,
+			Relation:  target.Relation,
+		},
+		SubjectIds: []string{target.ObjectId},
+		Metadata: &v1.ResolverMeta{
+			AtRevision:     revision.String(),
+			DepthRemaining: 50,
+		},
+	}, stream)
+	require.NoError(t, err)
+
+	for range stream.Results() {
+		// Break early
+		break
+	}
+	dispatcher.Close()
+}
+
+func TestReachableResourcesMultipleEntrypointEarlyCancel(t *testing.T) {
+	defer goleak.VerifyNone(t, goleakIgnores...)
+
+	rawDS, err := memdb.NewMemdbDatastore(0, 0, memdb.DisableGC)
+	require.NoError(t, err)
+
+	testRels := make([]*core.RelationTuple, 0)
+	for i := 0; i < 25; i++ {
+		testRels = append(testRels, tuple.MustParse(fmt.Sprintf("resource:res%d#viewer@user:tom", i)))
+		testRels = append(testRels, tuple.MustParse(fmt.Sprintf("resource:res%d#namespace@namespace:ns%d", i, i)))
+		testRels = append(testRels, tuple.MustParse(fmt.Sprintf("namespace:ns%d#parent@namespace:ns%d", i+1, i)))
+		testRels = append(testRels, tuple.MustParse(fmt.Sprintf("organization:org%d#member@user:tom", i)))
+		testRels = append(testRels, tuple.MustParse(fmt.Sprintf("namespace:ns%d#viewer@user:tom", i)))
+		testRels = append(testRels, tuple.MustParse(fmt.Sprintf("resource:someresource#org@organization:org%d", i)))
+	}
+
+	ds, revision := testfixtures.DatastoreFromSchemaAndTestRelationships(
+		rawDS,
+		`
+			definition user {}
+
+			definition organization {
+				relation direct_member: user
+				relation admin: user
+				permission member = direct_member + admin
+			}
+
+			definition namespace {
+				relation parent: namespace
+				relation viewer: user
+				relation admin: user
+				permission view = viewer + admin + parent->view
+			}
+
+			definition resource {
+				relation org: organization
+				relation admin: user
+				relation writer: user
+				relation viewer: user
+				relation namespace: namespace
+				permission view = viewer + writer + admin + namespace->view + org->member
+			}
+		`,
+		testRels,
+		require.New(t),
+	)
+	dispatcher := NewLocalOnlyDispatcher(2)
+
+	ctx := log.Logger.WithContext(datastoremw.ContextWithHandle(context.Background()))
+	require.NoError(t, datastoremw.SetInContext(ctx, ds))
```

Comment:
this seems extraneous to the test at first glance. Are you doing it so the datastore operations can be canceled since the context is not propagated to them and leaves goroutines hanging around?

Reply:
Its necessary to call the dispatcher; without it, it won't know the datastore to use

Reply:
oh whoops, you are right